Cython/Compiler/ExprNodes.py

```diff
@@ -503,21 +503,25 @@ def result(self):
         else:
             return self.calculate_result_code()
 
-    def _make_move_result_rhs(self, result, allow_move=True):
-        if (self.is_temp and allow_move and
-                self.type.is_cpp_class and not self.type.is_reference):
+    def _make_move_result_rhs(self, result, allow_move=True, force_move=False):
+        if force_move or (
```

Comment:
Sorry to add to the back-and-forth on this, but I think `optional=False` is a better choice here. Using `force_move=False` means that I have to doubly opt-in to get what I call. That seems wrong. When I say `move_result_rhs()`, it should generate a `move()` for the argument, unless that's really not possible (i.e. in non-C++ mode). I would even prefer an explicit (C?-)compile time error if I request a `move()` but it cannot be done. If I want it to be best-effort, then _that_ should be visible in the place where it's used. I should not have to jump through two levels of method calls to understand that the code that I'm looking at only sometimes does what it says.

Reply:
No worries. That makes sense -- I changed it to `optional=False`. What do you think about removing the `allow_move` kwarg here? I think the combinationn of `allow_move` and `optional` make the function hard to reason about. I propose instead of:

```
condition = ....
return _make_move_result_rhs(result, allow_move=condition, optional=optional)
```

We have:

```
condition = ...
return _make_move_result_rhs(result, optional=optional) if condition else result
```

Reply:
> removing the `allow_move` kwarg

Good idea. Why call the method when the caller already knows that it's not needed?

I cleaned up the methods, let's hope I didn't mess more things up than I improved.

Reply:
Thanks!